superset-frontend/src/explore/actions/saveModalActions.js

```diff
@@ -40,6 +40,17 @@ export function fetchDashboards(userId) {
           value: id,
           label: (json.result[index] || {}).dashboard_title,
         }));
+        choices.sort((a, b) => {
+          const labelA = a.label.toUpperCase();
+          const labelB = b.label.toUpperCase();
+          if (labelA < labelB) {
+            return -1;
+          }
+          if (labelA > labelB) {
+            return 1;
+          }
+          return 0;
```

Comment:
```suggestion
          return labelA.localeCompare(labelB);
```

Reply:
@michael-s-molina @geido Made a change so that when a string begins with a number it sorts numerically instead. Does not fix all the cases but addresses several.

Reply:
@agl-developer What you're trying to achieve is a concept called [Natural Sorting](https://en.wikipedia.org/wiki/Natural_sort_order). This is already supported natively by [localeCompare](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare) using `{ sensitivity: 'base', numeric: true }`